common/test/acceptance/tests/lms/test_conditional.py

```diff
@@ -0,0 +1,127 @@
+"""
+Bok choy acceptance tests for conditionals in the LMS
+"""
+from capa.tests.response_xml_factory import StringResponseXMLFactory
+from ..helpers import UniqueCourseTest
+from ...fixtures.course import CourseFixture, XBlockFixtureDesc
+from ...pages.lms.courseware import CoursewarePage
+from ...pages.lms.conditional import ConditionalPage, POLL_ANSWER
+from ...pages.lms.problem import ProblemPage
+from ...pages.studio.auto_auth import AutoAuthPage
+
+
+class ConditionalTest(UniqueCourseTest):
+    """
+    Test the conditional module in the lms.
+    """
+
+    def setUp(self):
+        super(ConditionalTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+        AutoAuthPage(
+            self.browser,
+            course_id=self.course_id,
+            staff=False
+        ).visit()
+
+    def install_course_fixture(self, block_type='problem'):
+        """
+        Install a course fixture
+        """
+        course_fixture = CourseFixture(
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run'],
+            self.course_info['display_name'],
+        )
+        vertical = XBlockFixtureDesc('vertical', 'Test Unit')
+        # populate the course fixture with the right conditional modules
+        course_fixture.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection').add_children(
+                    vertical
+                )
+            )
+        )
+        course_fixture.install()
+
+        # Construct conditional block
+        metadata = {}
+        source_block = None
+        if block_type == 'problem':
+            problem_factory = StringResponseXMLFactory()
+            problem_xml = problem_factory.build_xml(
+                question_text='The answer is "correct string"',
+                case_sensitive=False,
+                answer='correct string',
+            ),
+            problem = XBlockFixtureDesc('problem', 'Test Problem', data=problem_xml[0])
+            metadata = {
```

Comment:
Nit: rename as conditional_metadata? When I read this, I thought the metadata was related to the problem (then below, to the poll).